File: backend/internal/data/models/thread.go
package models

import (
	uuid "github.com/satori/go.uuid"
	"time"
)

type ThreadStatus string

const (
	ThreadScheduled  ThreadStatus = "SCHEDULED"
	ThreadProcessing ThreadStatus = "PROCESSING"
	ThreadFailed     ThreadStatus = "FAILED"
	ThreadDone       ThreadStatus = "DONE"
)

type Thread struct {
	BaseModel
	AccountID uuid.UUID `json:"-"`
	Account   *Account  `json:"account"`
	Tweets    []Tweet   `json:"tweets"`

	ScheduledFor time.Time    `json:"scheduled_for"`
	Status       ThreadStatus `json:"status"`
	Error        *string      `json:"error"`
}

File: frontend/src/api/entities/Thread.ts
import Tweet from "./Tweet";

export enum ThreadStatus {
    SCHEDULED = "SCHEDULED",
    PROCESSING = "PROCESSING",
    FAILED = "FAILED",
    DONE = "DONE",
}

type Thread = {
    id?: string,
    account: {
        id: string,
    },
    tweets: Tweet[],
    scheduled_for: Date,
    status: ThreadStatus,
    error: string|null,
}

export default Thread